capa/rules/__init__.py

```diff
@@ -1230,9 +1222,11 @@ def get_rules_and_dependencies(rules: List[Rule], rule_name: str) -> Iterator[Ru
     rules_by_name = {rule.name: rule for rule in rules}
     wanted = {rule_name}
 
-    def rec(rule):
+    def rec(rule: Rule):
         wanted.add(rule.name)
         for dep in rule.get_dependencies(namespaces):
+            if dep in wanted:
+                continue
```

Comment:
Adding `if` condtions should resolve circular dependencies among rules.

Reply:
I think we should raise an error here rather than swallowing the problem. Our rules should not support circular dependencies.